### PR TITLE
drivers licence number validation

### PR DESCRIPTION
**Pull request by lfdgov**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

Inputting a variety of DL number into citizen, staff portal and saving changes for valid DL numbers (7-9 digits for BC), (20 chars for others), required for Canada/US only

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/citizen-portal/src/app/services/notice-of-dispute.service.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class NoticeOfDisputeService {
     work_phone_number: [null, [FormControlValidators.phone]], // not using now
     email_address: [null, [Validators.required, Validators.email]],
     birthdate: [null, [Validators.required]],
-    drivers_licence_number: [null, [Validators.required]],
+    drivers_licence_number: [null, [Validators.required, Validators.minLength(7), Validators.maxLength(9)]],
```
Comment (Sharkww) [inline src/frontend/citizen-portal/src/app/services/notice-of-dispute.service.ts:35]: drivers_licence_province should be != "BC", it is better to be Validators.maxLength(20) or others
Maybe disable drivers_licence_number at first, enable it after drivers_licence_province is chosen

File: src/frontend/staff-portal/src/app/components/contact-info/contact-info.component.ts [hunk 2]
```diff
@@ -71,7 +71,7 @@ export class ContactInfoComponent implements OnInit {
       rejectedReason: [null, Validators.maxLength(256)], // Optional
       country: [null, [Validators.required]],
       postalCode: [null, [Validators.required, Validators.maxLength(6), Validators.minLength(6)]],
-      driversLicenceNumber: [null, [Validators.required, Validators.maxLength(20)]],
+      driversLicenceNumber: [null, [Validators.required, Validators.minLength(7), Validators.maxLength(9)]],
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/components/contact-info/contact-info.component.ts:74]: same as citizen portal

File: src/frontend/citizen-portal/src/app/components/dispute-ticket-stepper/dispute-ticket-stepper.component.ts [hunk 3]
```diff
@@ -184,7 +184,8 @@ export class DisputeTicketStepperComponent implements OnInit, AfterViewInit {
   public onDLProvinceChange(province) {
     setTimeout(() => {
       if (province == "BC") {
-        this.form.get("drivers_licence_number").setValidators([Validators.maxLength(9)])
+        this.form.get("drivers_licence_number").setValidators([Validators.maxLength(9)]);
+        this.form.get("drivers_licence_number").addValidators([Validators.minLength(7)]);
       } else {
```
Comment (Sharkww) [inline src/frontend/citizen-portal/src/app/components/dispute-ticket-stepper/dispute-ticket-stepper.component.ts:189]: Not necessary but can be setValidators([Validators.minLength(7), Validators.maxLength(9)])

**Review by Sharkww — CHANGES_REQUESTED** [review]

File: src/frontend/citizen-portal/src/app/services/notice-of-dispute.service.ts (hunk 1 above)
Comment (lfdgov) [reply src/frontend/citizen-portal/src/app/services/notice-of-dispute.service.ts:35]: as soon as dispute is retrieved, validators are reset

File: src/frontend/staff-portal/src/app/components/contact-info/contact-info.component.ts (hunk 2 above)
Comment (lfdgov) [reply src/frontend/staff-portal/src/app/components/contact-info/contact-info.component.ts:74]: as soon as dispute is retrieved validators are reset

File: src/frontend/citizen-portal/src/app/components/dispute-ticket-stepper/dispute-ticket-stepper.component.ts [hunk 4]
```diff
@@ -184,7 +184,8 @@ export class DisputeTicketStepperComponent implements OnInit, AfterViewInit {
...
         this.form.get("drivers_licence_number").setValidators([Validators.maxLength(20)]);
```
Comment (Sharkww) [inline src/frontend/citizen-portal/src/app/components/dispute-ticket-stepper/dispute-ticket-stepper.component.ts:190]: this.form.get("drivers_licence_number").setValidators([Validators.required, Validators.maxLength(20)]);

**Review by Sharkww — CHANGES_REQUESTED** [review]

**Review by Sharkww — APPROVED** [review]

File: src/frontend/citizen-portal/src/app/components/dispute-ticket-stepper/dispute-ticket-stepper.component.ts (hunk 4 above)
Comment (lfdgov) [reply src/frontend/citizen-portal/src/app/components/dispute-ticket-stepper/dispute-ticket-stepper.component.ts:190]: only required for Canada/US

**Pull request merged**